src/main/java/athleticli/commands/FindCommand.java

```diff
@@ -0,0 +1,40 @@
+package athleticli.commands;
+
+import java.time.LocalDate;
+import java.util.stream.Stream;
+
+import athleticli.commands.activity.FindActivityCommand;
+import athleticli.commands.diet.FindDietCommand;
+import athleticli.commands.sleep.FindSleepCommand;
+import athleticli.data.Data;
+import athleticli.exceptions.AthletiException;
+
+public class FindCommand extends Command {
+    protected LocalDate date;
+
+    public FindCommand(LocalDate date) {
+        this.date = date;
+    }
+
+    /**
+     * Returns the records to be shown to the user.
+     *
+     * @param data The current data.
+     * @return The records to be shown to the user.
+     * @throws AthletiException
+     */
+    @Override
+    public String[] execute(Data data) throws AthletiException {
+        try {
+            var activities = Stream.of(new FindActivityCommand(date).execute(data));
+            var diets = Stream.of(new FindDietCommand(date).execute(data));
+            var sleeps = Stream.of(new FindSleepCommand(date).execute(data));
+            return Stream.of(activities, diets, sleeps)
+                    .reduce(Stream::concat)
+                    .orElseGet(Stream::empty)
+                    .toArray(String[]::new);
+        } catch (AthletiException e) {
+            throw e;
+        }
```

Comment:
Unsure if the catch block here is intended as my understanding is that AthletiException will be caught by main AtheliCLI class to have the error message printed out.

Reply:
> Unsure if the catch block here is intended as my understanding is that AthletiException will be caught by main AtheliCLI class to have the error message printed out.

My bad. Originally, some exceptions were handled here, but later I improved the implementation of `FindCommand` to make it dependent on other find commands and then changed the exceptions in the `catch` block to `AthletiException` accordingly. I did not realize that actually I can just remove the entire `try` statement. 🤯